src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -21,8 +25,76 @@
     completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
     help='IoT Hub name.')
 
+dps_name_type = CLIArgumentType(
+    options_list=['--dps-name'],
+    completer=get_resource_name_completion_list('Microsoft.Devices/ProvisioningServices'),
+    help='IoT Provisioning Service name')
+
 
 def load_arguments(self, _):  # pylint: disable=too-many-statements
+    # Arguments for IoT DPS
+    with self.argument_context('iot dps') as c:
+        c.argument('dps_name', dps_name_type, options_list=['--name', '-n'], id_part='name')
+
+    with self.argument_context('iot dps create') as c:
+        c.argument('location', get_location_type(self.cli_ctx),
+                   help='Location of your IoT Provisioning Service. Default is the location of target resource group.')
+        c.argument('sku', arg_type=get_enum_type(IotDpsSku),
+                   help='Pricing tier for the IoT provisioning service.')
+        c.argument('unit', help='Units in your IoT Provisioning Service.', type=int)
+
+    for subgroup in ['access-policy', 'linked-hub', 'certificate']:
+        with self.argument_context('iot dps {}'.format(subgroup)) as c:
+            c.argument('dps_name', options_list=['--dps-name'], id_part=None)
+
+    with self.argument_context('iot dps access-policy') as c:
+        c.argument('access_policy_name', options_list=['--access-policy-name', '--name', '-n'],
+                   help='A friendly name for DPS access policy.')
+
+    with self.argument_context('iot dps access-policy create') as c:
+        c.argument('rights', options_list=['--rights', '-r'], nargs='+',
+                   arg_type=get_enum_type(AccessRightsDescription),
+                   help='Access rights for the IoT provisioning service. Use space separated list for multiple rights.')
+        c.argument('primary_key', help='Primary SAS key value.', type=str)
+        c.argument('secondary_key', help='Secondary SAS key value.', type=str)
+
+    with self.argument_context('iot dps access-policy update') as c:
+        c.argument('rights', options_list=['--rights', '-r'], nargs='+',
+                   arg_type=get_enum_type(AccessRightsDescription),
+                   help='Access rights for the IoT provisioning service. Use space separated list for multiple rights.')
+        c.argument('primary_key', help='Primary SAS key value.', type=str)
+        c.argument('secondary_key', help='Secondary SAS key value.', type=str)
```

Comment:
You never need to provide `type=str`.  That is always assumed to be the default.